semver.py

```diff
@@ -164,6 +164,9 @@ def __iter__(self):
         for v in self._astuple():
             yield v
 
+    def __getitem__(self, index):
+        return self._astuple()[index]
```

Comment:
Are we going to support negative index? e.g. `version_info[-3]`? In my opinion - no - so we should raise an `IndexError` exception if index is less than zero.

Reply:
It is not very hard to support negatives indices, as Python already provides this functionality with tuples. On the one hand, we can stay coherent with the reality of a version string in the way that we don't think of it in terms of a sequence, making negative indices unnecessary and incoherent.   

On the other hand, most python developers use negative indices and slices because this syntactic sugar is very convenient and intuitive in most cases. One could argue that using negative indices would be part of muscle memory in an experienced python developer's hands, thus allowing the use of negative indices more intuitive and natural.  

But the reality is that negative indices should be used only for sequences of variable length. Our version sequence length, in the state the project is now (correct me if I'm wrong), can only vary by 2 elements (prerelease and build number). *Can we consider this variable enough to necessitate negative indices?*

From what I gleaned by skimming over the issues, there are proposals to support incomplete version strings (like `major` or `major.minor`). If that were to be realised, negative indices could indeed become more useful. Unless other features are implemented that would make this redundant (a way to detect the *partialness* of a version that doesn't involve negative indices.)

Right now the path of least resistance is to allow the use of negative indices, since we are using a tuple internally to represent the version. In the future, we could remove support for this by adding the necessary checks, but that would break functionality that has been implemented, which would require a major version release according to semver.

I believe I laid out most of the arguments for and against negative indexing. In my opinion, we should go the path of least resistance, at the risk of having to break functionality in the future. I consider this risk relatively low, because negative indexing could become useful internally to this project.

I would like to know what the consensus is on this feature (or bug ;) ).

Reply:
Here are my 2 cents:

> But the reality is that negative indices should be used only for sequences of variable length. Our version sequence length, in the state the project is now (correct me if I'm wrong), can only vary by 2 elements (prerelease and build number). _Can we consider this variable enough to necessitate negative indices?_

IMHO, I think, that is the main argument against using negative indices. 

> Right now the path of least resistance is to allow the use of negative indices, since we are using a tuple internally to represent the version.

True, but from an OOP perspective, this is an implementation detail; the class should be perceived as a "black box". The user doesn't have to know how the class works internally.

---
What about the following strategy: If users doesn't ask for this specific feature, why implementing it? It there is no need, we can spare the work and focus on something different. Of course, we can come back, if people requesting this feature.

Also, **don't forget to document it** (regardless if this is implemented with or without negative indices).